holoviews/core/data/interface.py

```diff
@@ -50,22 +50,31 @@ def initialize(cls, eltype, data, kdims, vdims, datatype=None):
             kdims = pvals.get('kdims') if kdims is None else kdims
             vdims = pvals.get('vdims') if vdims is None else vdims
 
+        if datatype is None:
+            datatype = eltype.datatype
+
         # Process Element data
         if isinstance(data, NdElement):
             kdims = [kdim for kdim in kdims if kdim != 'Index']
-        elif hasattr(data, 'interface') and issubclass(data.interface, Interface):
-            data = data.data
+        elif (hasattr(data, 'interface') and issubclass(data.interface, Interface)):
+            if data.interface.datatype in datatype:
+                data = data.data
+            elif data.interface.gridded:
+                gridded = {kd.name: data.dimension_values(kd.name, expanded=False)
+                           for kd in data.kdims}
+                for vd in data.vdims:
+                    gridded[vd.name] = data.dimension_values(vd, flat=False)
+                data = gridded
```

Comment:
It looks like ``gridded`` the dictionary format for columns i.e a dictionary indexed by dimensions with array values. I wish there was a better name for this format as the variable ``gridded`` threw me off what it was for a while...